### ISSUE TITLE
Infinite loop parsing manifest

### ISSUE THREAD

**Issue by gregallen** [classgraph/classgraph#286]
parsing org.mortbay.jetty:jetty:4.0

gets stuck parsing the manifest at

nonapi.io.github.classgraph.fastzipfilereader.LogicalZipFile#parseManifest

debugger shows i=959
but length=960

its got stuck looking at the last two newlines but this loop doesn't run, since i is greater than 958:
```
if (skip) {
                // Field key didn't match -- skip to next key (after next newline that is not followed by a space)
                for (; i < manifest.length - 2;) {
```

**Comment by gregallen** [comment]
full manifest file is

```
Manifest-Version: 1.0
Sealed: true
Implementation-Version: 4.0.0
Specification-Version: 4.0
Package-Title: org.mortbay.jetty
Implementation-URL: http://jetty.mortbay.org
Package-Version: 1.0.0
Created-By: Ant 1.4.1
Implementation-Vendor: Mort Bay Consulting, Pty. Ltd.
Specification-Vendor: Mort Bay Consulting, Pty. Ltd.

Name: org/mortbay/http/
Specification-Title: HTTP Server
Specification-Version: 1.1
Implementation-Title: Mort Bay HTTP Server

Name: org/mortbay/jetty/servlet
Specification-Title:  J2EE Servlet Container
Specification-Version: 2.3
Implementation-Title: Jetty Servlet Container

Name: org/mortbay/jetty/
Specification-Title:  Jetty HTTP Servlet Server
Implementation-Title: Jetty HTTP Servlet Server

Name: org/mortbay/util/
Specification-Title:  Mort Bay Misc. Utilities
Implementation-Title: Mort Bay Misc. Utilities

Name: org/mortbay/html/
Specification-Title:  Mort Bay HTML Utilities
Implementation-Title: Mort Bay HTML Utilities

```

**Comment by lukehutch** [comment]
@gregallen Thank you for the detailed and helpful bug report! This is fixed and released in 4.6.3.

Sorry for the growing pains with the amount of new code in 4.6.
